File: 1260-day-of-the-year/day-of-the-year.java
class Solution {
    public int dayOfYear(String date) {
        String[] parts = date.split("-");
        int year = Integer.parseInt(parts[0]);
        int month = Integer.parseInt(parts[1]);
        int day = Integer.parseInt(parts[2]);

        int[] daysInMonth = {31,28,31,30,31,30,31,31,30,31,30,31};

        // Leap year check
        if ((year % 400 == 0) || (year % 4 == 0 && year % 100 != 0)) {
            daysInMonth[1] = 29;
        }

        int total = 0;
        for (int i = 0; i < month-1; i++) {
            total += daysInMonth[i];
        }

        return total + day;
    }
}

File: 1260-day-of-the-year/day-of-the-year.c
int dayOfYear(char * date){
    int year, month, day;
    sscanf(date, "%d-%d-%d", &year, &month, &day);
    
    int days_in_month[12] = {31,28,31,30,31,30,31,31,30,31,30,31};
    
    int isLeap = (year % 400 == 0) || (year % 4 == 0 && year % 100 != 0);
    if (isLeap) days_in_month[1] = 29;
    
    int total = 0;
    for(int i = 0; i < month-1; i++){
        total += days_in_month[i];
    }
    
    return total + day;
}
